src/lib/types.ts

```diff
@@ -16,21 +16,20 @@ import type {
 export type ChildOnlyProp = { children?: ReactNode }
 
 export type NextPageWithLayout<P = {}, IP = P> = NextPage<P, IP> & {
-  getLayout?: (page: ReactElement) => ReactNode
+  getLayout?: (page: ReactElement<P>) => ReactNode
```

Comment:
Pass generic used to define the props type to the page param so props extracted will be correctly typed when passed to layout components.